File: ios/ConectePhone/SIP/SipEngine.swift
import Combine
import Foundation

@MainActor
protocol SipEngine: AnyObject {
    var registration: RegistrationState { get }
    var activeCall: ActiveCall? { get }
    var changes: AnyPublisher<Void, Never> { get }

    func register(account: SIPAccount, username: String, password: String) async throws
    func unregister() async
    func call(_ handle: String) async throws
    func prepareIncoming(id: UUID, displayName: String, handle: String)
    func answer(id: UUID) async throws
    func reject(id: UUID) async
    func end() async
    func setMuted(_ value: Bool)
    func setSpeaker(_ value: Bool)
    func sendDTMF(_ digit: Character)
}

File: android/app/src/main/java/br/com/conectemax/phone/sip/SipEngine.kt
package br.com.conectemax.phone.sip

import br.com.conectemax.phone.model.ActiveCall
import br.com.conectemax.phone.model.RegistrationState
import br.com.conectemax.phone.model.SipAccount
import kotlinx.coroutines.flow.StateFlow

interface SipEngine {
    val registration: StateFlow<RegistrationState>
    val activeCall: StateFlow<ActiveCall?>
    suspend fun register(account: SipAccount, username: String, password: CharArray)
    suspend fun unregister()
    fun updatePushToken(token: String)
    suspend fun call(handle: String)
    fun prepareIncomingPush(callId: String, displayName: String, handle: String)
    suspend fun answer(callId: String)
    suspend fun reject(callId: String)
    suspend fun end()
    fun setMuted(muted: Boolean)
    fun setSpeaker(enabled: Boolean)
    fun sendDtmf(digit: Char)
}
